src/http.cc

```diff
@@ -2178,13 +2363,17 @@ HttpStateData::buildRequestPrefix(MemBuf * mb)
     /* build and pack headers */
     {
         HttpHeader hdr(hoRequest);
+        makeUpgradeHeaders(hdr);
         httpBuildRequestHeader(request.getRaw(), entry, fwd->al, &hdr, flags);
 
         if (request->flags.pinned && request->flags.connectionAuth)
             request->flags.authSent = true;
         else if (hdr.has(Http::HdrType::AUTHORIZATION))
             request->flags.authSent = true;
 
+        if (hdr.has(Http::HdrType::UPGRADE))
+            upgradeHeaderOut = new String(hdr.getList(Http::HdrType::UPGRADE));
```

Comment:
commit 3cc1de134a

It is always nil. The outgoing headers build once in HttpStateData::start.
Probably instead of assertion just a "delete upgradeHeaderOut" was better just in case in the future we are doing more than one tries to build headers. Anyway.